folding-schemes/src/folding/traits.rs

```diff
@@ -0,0 +1,121 @@
+use ark_crypto_primitives::sponge::{
+    constraints::{AbsorbGadget, CryptographicSpongeVar},
+    poseidon::constraints::PoseidonSpongeVar,
+    Absorb,
+};
+use ark_ec::CurveGroup;
+use ark_ff::PrimeField;
+use ark_r1cs_std::{alloc::AllocVar, fields::fp::FpVar, ToConstraintFieldGadget};
+use ark_relations::r1cs::SynthesisError;
+
+use crate::{transcript::Transcript, Error};
+
+use super::circuits::CF1;
+
+pub trait CommittedInstanceExt<C: CurveGroup> {
+    /// The in-circuit representation of the committed instance.
+    type Var: AllocVar<Self, CF1<C>> + CommittedInstanceVarExt<C>;
+    /// `hash` implements the committed instance hash compatible with the
+    /// in-circuit implementation from `CommittedInstanceVarOps::hash`.
```

Comment:
is it `CommittedInstanceVarOps` or `CommittedInstanceVarExt`?

Reply:
Nice question! I thought a bit about the naming of these traits and chose `Ext` because I saw it in some APIs provided by rust std, e.g., the [`FileExt`](https://doc.rust-lang.org/std/os/unix/fs/trait.FileExt.html) trait specifies methods that are going to be implemented in the [`File`](https://doc.rust-lang.org/std/fs/struct.File.html) struct.
I have also considered `Ops`, but I couldn't remember why I preferred `Ext` over it 😂

Now I realize that `Ext` may sound like it is for *extending* committed instances and witnesses, but these traits in fact have *core* methods that every struct should implement. So `Ops` should be a better choice!

Reply:
Thanks for the context. From my side I'm fine with `Ext`, makes sense.
Here in the comment I was referring to the in-code comment using `Ops` while the type everywhere else is `Ext`, so that the comment refers to an unexisting type.

Reply:
Oops I didn't notice the `CommittedInstanceVarOps::hash` part and have already renamed `Ext` to `Ops`... Should I revert the commit?

Reply:
Up to you, I'll be happy either way hehe. The original comment was just to point out the in-code comment not matching the used name. But happy with any of both names, so whatever one you prefer :)